app/assets/javascripts/host_checkbox.js

```diff
@@ -97,19 +97,19 @@ function cleanHostsSelection() {
 
 function multiple_selection() {
   var total = $("#pagination").data("count");
-  var alert_text = Jed.sprintf(__("All <b> %d </b> hosts are selected. "), total);
+  var alert_text = Jed.sprintf(__("All <b> %d </b> hosts are selected."), total);
   var undo_text = __("Undo selection");
   var multiple_alert = $("#multiple-alert");
-  multiple_alert.find(".text").html(alert_text + '<a href="#" onclick="undo_multiple_selection();">' + undo_text + '</a>');
+  multiple_alert.find(".text").html(alert_text + '<a href="#" onclick="undo_multiple_selection();">' + ' ' + undo_text + '</a>');
```

Comment:
please don't add a space inside the link, it will cause an extra underline before the text. If you want to make sure that a space is added you can use `&nbsp;`, otherwise html will shrink any spaces added in the code. 
Also no need to concatenate a space to the tag, you can add it inside the existing string.